File: Player.cpp
/*
 * Noam shimshoviz 203565429 Sarit Zevin 313242588
 */

#include "Player.h"


Player::Player(discSymbol p) {
	player = p;
	if(player == X) {
		other = O;
	}
	else {
		other = X;
	}
}

discSymbol Player::getOther() const {
	return other;
}

discSymbol Player::getPlayer() const {
	return player;
}

vector<Position> Player::hasMove(GameLogic logic) const {
	const vector<Position>& moves = logic.calculateMoves(this->getPlayer());
	return moves;

}

File: javaFXEx6/Player.java
package javaFXEx6;
/*
 * Noam shimshoviz 203565429 Sarit Zevin 313242588
 */
public class Player {
	private char player;
	private char other;
	public Player(char p) {
		this.player = p;
		if(player == 'X') {
			this.other = 'O';
		}
		else {
			this.other = 'X';
		}
	}

	public char getOther() {
		return other;
	}

	public char getPlayer() {
		return player;
	}


}
